runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -103,7 +103,16 @@ function TSHighlighter:get_hl_from_capture(capture)
   else
     -- Default to false to avoid recomputing
     local hl = TSHighlighter.hl_map[name]
-    return hl and a.nvim_get_hl_id_by_name(hl) or 0
+
+    if not hl then return 0 end
+
+    local scoped_hl = vim.fn.hlID(self.parser.lang .. hl)
```

Comment:
hmm Treesitter is a bit of a special case but in general it is considered bad style to check for the existence of highlight groups like this. What typically should be done is `hi default link PythonComment Comment` (or an API equivalent that I could introduce). This command is a no-op if `PythonComment` already exist or a default has been defined elsewhere.

Reply:
concretely, even if `PythonComment` does not exist the _metadata_ that it was a PythonComment will be preserved for downstream consumers (like hlstate ui:s and the yet nonexisting libnvim)

Reply:
Yeah okay I see, then that's the appropriate time to introduce treesitter highlights

Reply:
I don't understand what you're saying @vigoux .

What we should do is like bfredl said, you always apply `PythonTSComment`, which, by default, is linked to `TSComment` which can be linked to something else if desired. I think this will be good (and it was what I was asking for in the original issue, so that would make me happy :laughing: )

Reply:
Yeah that's what I meant, sorry if I wasn't clear 😉